src/components/Card.js

```diff
@@ -1,25 +1,43 @@
 class Card {
-  constructor(data, templateSelector, handleImageClick) {
+  constructor(
+    data,
+    templateSelector,
+    handleImageClick,
+    handleDeleteButton,
+    handleLike
+  ) {
     this._name = data.name;
     this._link = data.link;
     this._templateSelector = templateSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteButton = handleDeleteButton;
+    this._id = data._id;
+    this._handleLike = handleLike;
+    this._isLiked = false;
   }
 
   _setEventListeners() {
     this._likeButton = this._element.querySelector(".card__like-button");
     this._deleteButton = this._element.querySelector(".card__remove-button");
-    this._likeButton.addEventListener("click", () => this._handleLikeButton());
-    this._deleteButton.addEventListener("click", () =>
-      this._handleDeleteCard()
-    );
+    this._likeButton.addEventListener("click", () => {
+      if (this._likeButton.classList.contains("card__like-button_active")) {
+        this._handleLike(this.getId());
+        this._likeButton.classList.remove("card__like-button_active");
```

Comment:
Needs correction: please don't change the likes button state here, you should do that only after the successful api response.

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

isLiked() {
  return this._isLiked;
}

_renderLikes() {
  if (this._isLiked) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}

createCard() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  api.dislikeCard(...).then(response => card.setIsLiked(response._isLiked)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```